include/cxxopts.hpp

```diff
@@ -755,19 +778,23 @@ inline bool IsFalseText(const std::string &text)
   return !result.empty();
 }
 
-inline std::pair<std::string, std::string> SplitSwitchDef(const std::string &text)
+// Gets the option names specified via a single, comma-separated string,
+// and returns the separate, space-discarded, non-empty names
+// (without considering which or how many are single-character)
+inline OptionNames split_option_names(const std::string &text)
 {
-  std::match_results<const char*> result;
-  std::regex_match(text.c_str(), result, option_specifier);
-  if (result.empty())
+  if (not std::regex_match(text.c_str(), option_specifier))
   {
     throw_or_mimic<exceptions::invalid_option_format>(text);
   }
 
-  const std::string& short_sw = result[2];
-  const std::string& long_sw = result[3];
+  OptionNames split_names;
 
-  return std::pair<std::string, std::string>(short_sw, long_sw);
+  constexpr const int use_non_matches { -1 };
```

Comment:
I think the extra `const` is redundant.

Reply:
@jarro2783 : Only in C++11. If you compile with C++14 or later, `const` isn't implied. I don't mind very much, the only reason I have this is to avoid using `-1` as a "magic number".

Reply:
Ok I didn't realise that, this is fine then.

Reply:
Actually I think you've misread the statement about const.
> A constexpr specifier used in an object declaration [or non-static member function (until C++14)] implies const

It is only non-static member functions that were dropped from implying const. So this extra `const` is redundant.

Reply:
Well, that's counter-intuitive, but ok. Dropped the const.